### ISSUE TITLE
fails to build with Fedora 40 (gcc-14)

### ISSUE THREAD

**Issue by juhp** [haskell-cryptography/HsOpenSSL#88]
Downstream bug is here: https://bugzilla.redhat.com/show_bug.cgi?id=2261128

It fails on all Fedora archs: https://koji.fedoraproject.org/koji/taskinfo?taskID=115236336

The [x86_64 log](https://kojipkgs.fedoraproject.org/work/tasks/8476/115238476/build.log) looks like:
[build.log](https://github.com/haskell-cryptography/HsOpenSSL/files/14692140/build.log)

**Comment by istathar** [comment]
Does this mean we're going to have to add 

```
ghc-options:
  HsOpenSSL: -optc=-Wno-incompatible-pointer-types
```

to every project that transitively depends on HsOpenSSL?

**Comment by Kleidukos** [comment]
Yeah, `CApiFFI` needs https://hackage.haskell.org/package/base-4.19.1.0/docs/Foreign-C-ConstPtr.html but you may stumble upon https://gitlab.haskell.org/ghc/ghc/-/issues/24814, although the chances for this problem to happen are not very high.

**Comment by erikd** [comment]
As per the comment https://github.com/haskell-cryptography/HsOpenSSL/pull/94#issuecomment-2297487546 , `0.11.7.8` has been released to Hackage but still fails to build with `ghc-9.8.2` on Linux.

**Comment by sternenseemann** [comment]
For the background of this failure see the [Porting to GCC 14 guide](https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors). GCC 14 behaves as if `-Werror=incompatible-pointer-types` was passed.

**Comment by erikd** [comment]
See https://github.com/haskell-cryptography/HsOpenSSL/pull/102

For packages that depend on `HsOpenSSL` the following works (for now):
```
package HsOpenSSL
  ghc-options: -optc=-Wno-discarded-qualifiers -optc=-Wno-deprecated-declarations -optc=-Wno-incompatible-pointer-types
```
